SCons/Environment.py

```diff
@@ -1986,11 +1988,14 @@ def Builder(self, **kw):
         nkw = self.subst_kw(kw)
         return SCons.Builder.Builder(**nkw)
 
-    def CacheDir(self, path):
+    def CacheDir(self, path, custom_class=None):
         if path is not None:
             path = self.subst(path)
         self._CacheDir_path = path
 
+        if custom_class:
+            self['CACHEDIR_CLASS'] = custom_class
```

Comment:
Shouldn't this check here if the class is indeed a child of the CacheDir class?

Reply:
yeah, i'll update

Reply:
fixed in ce9ea25